assets/js/theme/common/carousel/utils/handleImageAspectRatio.js

```diff
@@ -32,15 +40,11 @@ export default ({ delegateTarget }, carousel) => {
 
     if ($activeSlide.find('.heroCarousel-content').length) return;
 
-    $('<img />')
-        .on('load', function getImageSizes() {
-            const imageHeight = this.height;
-            const imageWidth = this.width;
-
-            if (imageHeight < 2 || imageWidth < 2) return;
-
-            const imageAspectRatio = imageHeight / imageWidth;
-            $activeSlideAndClones.each((idx, slide) => $(slide).addClass(defineClass(imageAspectRatio)));
-        })
-        .attr(attrsObj);
+    if (activeSlideImgNode.complete) {
+        if (activeSlideImgNode.naturalHeight === 1) {
```

Comment:
Could you add some explanation comments here?

Reply:
If image is loaded but (for some reasons) it still has not its own real height, we can not define is image portrait or landscape and it will be updated next time when slide will be active

Reply:
I mean add some comments into the code)

Reply:
&#x2705;